Documentation/ApiOverview/RequestHandling/Index.rst

```diff
@@ -0,0 +1,271 @@
+
+Enriching response
+-----------------
+
+This middleware will check the length of generated output, and add an header
+with this information to the response.
+
+In order to do so, next request handler is called. He will return the generated
+response, which can be enriched before it gets returned.
+
+::
+
+   public function process(
+       ServerRequestInterface $request,
+       RequestHandlerInterface $handler
+   ): ResponseInterface {
+       $response = $handler->handle($request);
+
+       if (/* if logic */) {
+           $response = $response->withHeader(
+               'Content-Length',
+               (string)$response->getBody()->getSize()
+           );
+       }
+
+       return $response;
+   }
+
+.. _request-handling-configuring-middlewares:
+
+Configuring middlewares
+=======================
+
+In order to implement a custom middleware, this middleware has to be configured.
+TYPO3 already provides some middlewares pre configured. Beside adding own
+middlewares, it's also possible to remove existing middlewares from
+configuration.
+
+The configuration is provided within
+:file:`Configuration/RequestMiddlewares.php` of an extensions::
+
+   return [
+       'frontend' => [
+           'middleware-identifier' => [
+               'target' => \Vendor\ExtName\Middleware\ConcreteClass::class,
+               'before' => [
+                   'another-middleware-identifier',
+               ],
+               'after' => [
+                   'yet-another-middleware-identifier',
+               ],
+           ],
+       ],
+       'backend' => [
+           'middleware-identifier' => [
+               'target' => \Vendor\ExtName\Middleware\AnotherConcreteClass::class,
+               'before' => [
+                   'another-middleware-identifier',
+               ],
+               'after' => [
+                   'yet-another-middleware-identifier',
+               ],
+           ],
+       ],
+   ];
+
+TYPO3 has multiple stacks where one middleware might only be necessary in one
+of them. Therefore the configuration first introduces an level to define the
+context. Within each context each middleware is registered as new php array with
+an unique identifier as key.
+
+Possible stacks are: `frontend` and `backend`.
+
+Each middleware consists of the following options:
+
+target
+   PHP string
+
+   FQCN (=Fully Qualified Class Name) to use as middleware.
+
+before
+   PHP Array
+
+   List of middleware identifiers. The middleware itself is
+   executed before any other middleware within this array.
+
+after
+   PHP Array
+
+   List of middleware identifiers. The middleware itself is
+   executed after any other middleware within this array.
+
+disabled
+   PHP boolean
+
+   Allows to disable specific middlewares.
+
+.. _request-handling-debugging:
+
+Debugging
+=========
+
+In order to see which middlewares are configured, and to see the order of
+execution TYPO3 offers a new function menu entry within "Configuration" module:
```

Comment:
"execution, TYPO3" and "new" should be dropped. Also the menu entry should be named here.